sktime/classifiers/shapelet_based/mrseql/mrseql.pyx

```diff
@@ -0,0 +1,439 @@
+            m = PySEQL()
+            m.learn(sequences, tmp_labels)
+            self.models.append(m)
+
+    def classify(self, string sequence):
+        scr = []
+        for m in self.models:
+            scr.append(m.classify(sequence)[1])
+        return np.array(scr)
+
+    def get_sequence_features(self, bool only_positive = False):
+        sqs = []
+        for m in self.models:
+            sqs.extend(m.get_sequence_features(True))
+        return sqs
+
+    def get_coefficients(self, bool only_positive = False):
+        coefs = []
+        for m in self.models:
+            coefs.extend(m.get_coefficients(True))
+        return coefs
+
+
+###########################################################################
+
+
+######################### Mr-SEQL (main class) #########################
+''' Time Series Classification with multiple symbolic representations and SEQL (Mr-SEQL)
+
+ @article{mrseql,
+ author = {Le Nguyen, Thach and Gsponer, Severin and Ilie, Iulia and O'reilly, Martin and Ifrim, Georgiana},
+ title = {Interpretable Time Series Classification Using Linear Models and Multi-resolution Multi-domain Symbolic Representations},
+ journal = {Data Mining and Knowledge Discovery},
+ volume = {33},
+ number = {4},
+ year = {2019},
+ }
+
+
+Overview: Mr-SEQL is a time series classifier that learn from multiple symbolic representations of multiple resolutions and multiple domains.
+Currently, Mr-SEQL supports both SAX and SFA representations.
+
+Parameters
+----------
+seql_mode       : str, either 'clf' or 'fs'. In the 'clf', Mr-SEQL mode trains an ensemble of SEQL models while in the 'fs' mode it uses SEQL to select features for training a logistic regression model.
+symrep          : list or tuple, should contains only 'sax' or 'sfa' or both. The symbolic representations to be used to transform the input time series.
+symrep_config   : dict, customized parameters for the symbolic transformation. If defined, symrep will be ignored.
+
+'''
+class MrSEQLClassifier(BaseClassifier):
+
+
+    def __init__(self, seql_mode='clf', symrep=['sax'], symrepconfig=None):
+
+        self.symbolic_methods = symrep
+
+        if seql_mode in ('fs','clf'):
+            self.seql_mode = seql_mode
+        else:
+            raise ValueError('seql_mode should be either clf or fs.')
+
+        if symrepconfig is None:
+            self.config = [] # http://effbot.org/zone/default-values.htm
+        else:
+            self.config = symrepconfig
+
+        #self.label_dict = {} # for translating labels since seql only accept [1.-1] as labels
+        self.seql_models = [] # seql models
+
+        # all the unique labels in the data
+        # in case of binary data the first one is always the negative class
+        self.classes_ = []
+
+
+
+        self.clf = None # scikit-learn model
+
+        # store fitted sfa for later transformation
+        self.sfas = {}
+
+
+    def __is_multiclass(self):
```

Comment:
Please don't use double underscores here and throughout, they have a different meaning in [Python](https://stackoverflow.com/questions/1301346/what-is-the-meaning-of-a-single-and-a-double-underscore-before-an-object-name) and [scikit-learn](https://stackoverflow.com/a/16437327/9334962), single one will be just fine.